Partial Update Support - For Non-OpenGL platform  and OpenGl platform without swapbufferWithDamage extension

munezbn opened the pull request:
Added support for Non-OpenGL platform and OpenGL platform who doesn't have extension for swapbuffer with damage.
Added API to detect platforms capabilities for  "swapbuffer _with_damage_rect" and "copy frontbuffer to backbuffer"
Added Offscreen FBO draw for supporting partial updates on OpenGL platform who doesn't have extension for swapbuffer with damage.

munezbn committed:
Partial Update Support - Part 4
Added support for Non-OpenGL platform and OpenGL platform who doesnt have extension for swapbuffer with damage.
Added API to detect platforms capabilities for  "swapbuffer_with_damage_rect" and "copy frontbuffer to backbuffer"

munezbn committed:
Minor Improvement in onHasCopyBuffer() function

pprabhu-ng commented on rns_shell/platform/graphics/gl/egl/GLWindowContextEGL.cpp:
```diff
@@ -438,10 +452,20 @@ std::vector<EGLint> GLWindowContextEGL::RectsToInts(EGLDisplay display, EGLSurfa
 }
 
+bool GLWindowContextEGL::onHasBufferCopy() {
+#if USE(RNS_SHELL_PARTIAL_UPDATES) && USE(RNS_SHELL_COPY_BUFFERS)
```
RNS_SHELL_PARTIAL_UPDATES check seems to be already done at line 455.

munezbn replied:
Addressed

munezbn committed:
Added partial update support for GPU platform which doesnt have eglSwapBufferWithDamage extentions.

Will use offscreen FBO to draw and then Blit it to main FB followed by eglswapbuffers call.
NOTE: This is an experimental feature which uses glBlitFramebuffer to copy offscreen FBO to Main FB.
      This may not be performant on some hardware so use it accordingly.
      This is disabled by default. define "USE_RNS_SHELL_COPY_BUFFERS 1" to enable it

munezbn committed:
Fixed OpenGl ES build fail due to invalid header
